code/modules/antagonists/xeno/xeno.dm

```diff
@@ -31,6 +31,54 @@
 /datum/antagonist/xeno/get_team()
 	return xeno_team
 
+/mob/living/carbon/alien/humanoid/royal/queen/Login()
+	..()
+	SSshuttle.registerHostileEnvironment(src)
+
+/mob/living/carbon/alien/humanoid/royal/queen/Logout()
+	..()
+	addtimer(CALLBACK(src, .proc/logoutcheck), 12 SECONDS)//when logging out, we don't want to spam announcements with leaving and reconnecting.
```

Comment:
should be 30 seconds at least because thats the timer not byond premium people have to wait to reconnect